tests/scenarios/nested-roles.spec.ts

```diff
@@ -0,0 +1,281 @@
+      expect(channel2RecordReply.status.code).to.equal(202);
+
+      // 5. Bob as the creator/author of the channels can  add participants in the gated-channels
+      // 5a. Bob can add himself and Carol as participants in the gated-channel 1
+      const channel1ParticipantBobRecord = await TestDataGenerator.generateRecordsWrite({
+        author          : bob,
+        recipient       : bob.did,
+        protocol        : protocolDefinition.protocol,
+        protocolPath    : 'community/gatedChannel/participant',
+        parentContextId : channel1Record.message.contextId,
+      });
+      const channel1ParticipantBobRecordReply
+        = await dwn.processMessage(alice.did, channel1ParticipantBobRecord.message, { dataStream: channel1ParticipantBobRecord.dataStream });
+      expect(channel1ParticipantBobRecordReply.status.code).to.equal(202);
+
+      const channel1ParticipantCarolRecord = await TestDataGenerator.generateRecordsWrite({
+        author          : bob,
+        recipient       : carol.did,
+        protocol        : protocolDefinition.protocol,
+        protocolPath    : 'community/gatedChannel/participant',
+        parentContextId : channel1Record.message.contextId,
+      });
+      const channel1ParticipantCarolRecordReply
+        = await dwn.processMessage(alice.did, channel1ParticipantCarolRecord.message, { dataStream: channel1ParticipantCarolRecord.dataStream });
+      expect(channel1ParticipantCarolRecordReply.status.code).to.equal(202);
+
+      // 5b. Bob can add himself and Daniel as participants in the gated-channel 2
+      const channel2ParticipantBobRecord = await TestDataGenerator.generateRecordsWrite({
+        author          : bob,
+        recipient       : bob.did,
+        protocol        : protocolDefinition.protocol,
+        protocolPath    : 'community/gatedChannel/participant',
+        parentContextId : channel2Record.message.contextId,
+      });
+      const channel2ParticipantBobRecordReply
+              = await dwn.processMessage(alice.did, channel2ParticipantBobRecord.message, { dataStream: channel2ParticipantBobRecord.dataStream });
+      expect(channel2ParticipantBobRecordReply.status.code).to.equal(202);
+
+      const channel2ParticipantDanielRecord = await TestDataGenerator.generateRecordsWrite({
+        author          : bob,
+        recipient       : daniel.did,
+        protocol        : protocolDefinition.protocol,
+        protocolPath    : 'community/gatedChannel/participant',
+        parentContextId : channel2Record.message.contextId,
+      });
+      const channel2ParticipantDanielRecordReply
+        = await dwn.processMessage(alice.did, channel2ParticipantDanielRecord.message, { dataStream: channel2ParticipantDanielRecord.dataStream });
+      expect(channel2ParticipantDanielRecordReply.status.code).to.equal(202);
+
+      // 6. Carol CANNOT add anyone as a participant in the gated-channel 2 since she is not a participant in the channel
```

Comment:
Not sure if it's relevant to add the participant test case as well?

```suggestion
      // 6. Carol send a message to the gated-channel 2 since she is not a participant in the channel
```

Reply:
Hey @LiranCohen, the comment should be correct, it is testing to make sure Carol can't add any participant (note the expected 401 code).

Let me know if I am not making sense!

Reply:
Am I misreading it?  Below I see `carolChatMessage` which has `protocolPath` of `'community/gatedChannel/message'`.

Seems like she is trying to add a message to the chat using a participant role she doesn't have.

Reply:
No no, you are right, the intend was correct, but my code was wrong! Fixing now!

Reply:
Ok SUPER small nit lol, `carolChatMessage` should probably be renamed.